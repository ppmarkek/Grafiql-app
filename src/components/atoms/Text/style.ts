import { Typography } from '@mui/material';
import { styled } from '@mui/system';

type TextStyleProps = {
  $small?: boolean;
  $color?: string;
};

export const H1 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '34px',
  lineHeight: '42px',
  color: $color || '#000',
}));

export const H2 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '28px',
  lineHeight: '38px',
  color: $color || '#000',
}));

export const H3 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '25px',
  lineHeight: '32px',
  color: $color || '#000',
}));

export const H4 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '23px',
  lineHeight: '27px',
  color: $color || '#000',
}));

export const H5 = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color',
})(({ $color }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: '21px',
  lineHeight: '24px',
  color: $color || '#000',
}));

export const Bold = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color' && prop !== '$small',
})(({ $color, $small }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '700',
  fontSize: $small ? '16px' : '18px',
  color: $color || '#000',
  lineHeight: '21px',
}));

export const Regular = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$color' && prop !== '$small',
})(({ $color, $small }: TextStyleProps) => ({
  fontFamily: 'Heebo, sansSerif',
  fontStyle: 'normal',
  fontWeight: '400',
  fontSize: $small ? '16px' : '18px',
  lineHeight: '21px',
  color: $color || '#000',
}));

export const Error = styled(Typography, {
  shouldForwardProp: (prop) => prop !== '$small',
})(({ $small }: TextStyleProps) => ({
  color: 'rgb(211, 47, 47)',
  fontFamily: 'Heebo, sansSerif',
  fontWeight: '400',
  fontSize: $small ? '16px' : '18px',
  lineHeight: '21px',
}));
