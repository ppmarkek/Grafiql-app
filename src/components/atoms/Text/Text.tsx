import { Bold, Error, H1, H2, H3, H4, H5, Regular } from './style';

enum Variant {
  'H1',
  'H2',
  'H3',
  'H4',
  'H5',
  'BOLD',
  'REGULAR',
  'ERROR',
}

type TextProps = {
  variant:
    | 'H1'
    | 'H2'
    | 'H3'
    | 'H4'
    | 'H5'
    | 'BOLD'
    | 'REGULAR'
    | 'ERROR'
    | Variant;
  small?: boolean;
  children?: string | number | React.ReactNode;
  color?: string;
};

const Text = ({ variant, small, children, color }: TextProps) => {
  switch (variant) {
    case Variant[Variant.H1]:
      return <H1 $color={color}>{children}</H1>;
    case Variant[Variant.H2]:
      return <H2 $color={color}>{children}</H2>;
    case Variant[Variant.H3]:
      return <H3 $color={color}>{children}</H3>;
    case Variant[Variant.H4]:
      return <H4 $color={color}>{children}</H4>;
    case Variant[Variant.H5]:
      return <H5 $color={color}>{children}</H5>;
    case Variant[Variant.BOLD]:
      return (
        <Bold $small={small} $color={color}>
          {children}
        </Bold>
      );
    case Variant[Variant.REGULAR]:
      return (
        <Regular $small={small} $color={color}>
          {children}
        </Regular>
      );
    case Variant[Variant.ERROR]:
      return <Error $color={color}>{children}</Error>;
    default:
      return <Regular $small={small}>{children}</Regular>;
  }
};

export default Text;
