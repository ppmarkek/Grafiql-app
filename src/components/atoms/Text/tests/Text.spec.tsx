import React from 'react';
import { render } from '@testing-library/react';
import Text from '../Text';

enum Variant {
  'H1',
  'H2',
  'H3',
  'H4',
  'H5',
  'BOLD',
  'REGULAR',
  'ERROR',
}

describe('Text Component', () => {
  const variants: (keyof typeof Variant)[] = [
    'H1',
    'H2',
    'H3',
    'H4',
    'H5',
    'BOLD',
    'REGULAR',
    'ERROR',
  ];

  variants.forEach((variant) => {
    it(`renders ${variant} variant correctly`, () => {
      const { asFragment } = render(
        <Text variant={variant}>{`${variant} Text`}</Text>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders with custom color', () => {
    const { asFragment } = render(
      <Text variant="H1" color="blue">
        Colored Header
      </Text>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders small text when small prop is true', () => {
    const { asFragment } = render(
      <Text variant="BOLD" small>
        Small Bold Text
      </Text>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders regular text by default', () => {
    const { asFragment } = render(<Text variant={'H1'}>Default Text</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with numerical children', () => {
    const { asFragment } = render(<Text variant="H1">12345</Text>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no children', () => {
    const { asFragment } = render(<Text variant="H1" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
