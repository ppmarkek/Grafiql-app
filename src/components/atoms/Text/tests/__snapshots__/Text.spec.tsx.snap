// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders BOLD variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-p2dihp-MuiTypography-root"
  >
    BOLD Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders ERROR variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-yypkrf-MuiTypography-root"
  >
    ERROR Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders H1 variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-u9cgmb-MuiTypography-root"
  >
    H1 Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders H2 variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-xgonna-MuiTypography-root"
  >
    H2 Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders H3 variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-3k5gu6-MuiTypography-root"
  >
    H3 Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders H4 variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-sl9dv0-MuiTypography-root"
  >
    H4 Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders H5 variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-108gikc-MuiTypography-root"
  >
    H5 Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders REGULAR variant correctly 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-55cgef-MuiTypography-root"
  >
    REGULAR Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders correctly with no children 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-u9cgmb-MuiTypography-root"
  />
</DocumentFragment>
`;

exports[`Text Component renders correctly with numerical children 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-u9cgmb-MuiTypography-root"
  >
    12345
  </p>
</DocumentFragment>
`;

exports[`Text Component renders regular text by default 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-u9cgmb-MuiTypography-root"
  >
    Default Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders small text when small prop is true 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-118wq2h-MuiTypography-root"
  >
    Small Bold Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders with custom color 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-93oes6-MuiTypography-root"
  >
    Colored Header
  </p>
</DocumentFragment>
`;
